src/server/terrain/structure_building_blocks.zig

```diff
@@ -111,59 +104,50 @@ pub fn isOriginBlock(block: Block) bool {
 	return block.typ == originBlockNumericId;
 }
 
-const originBlockStringId = "cubyz:sbb/origin";
-var originBlockNumericId: u16 = 0;
-
-// Maps global child block numeric ID to index used to locally represent that child block.
-var childBlockNumericIdMap: std.AutoHashMapUnmanaged(u16, u16) = .{};
-var childBlockStringId: ListUnmanaged([]const u8) = .{};
-
 pub const StructureBuildingBlock = struct {
 	children: []AliasTable(Child),
-	blueprint: [4]*BlueprintEntry,
+	blueprints: [4]BlueprintEntry,
 
 	fn initFromZon(stringId: []const u8, zon: ZonElement) !StructureBuildingBlock {
-		const blueprintId = zon.get(?[]const u8, "blueprint", null);
-		if(blueprintId == null) {
+		const blueprintId = zon.get(?[]const u8, "blueprint", null) orelse {
 			std.log.err("['{s}'] Missing blueprint field.", .{stringId});
 			return error.MissingBlueprintIdField;
-		}
-		const blueprintRef = blueprintCache.get(blueprintId.?);
-		if(blueprintRef == null) {
-			std.log.err("['{s}'] Could not find blueprint '{s}'.", .{stringId, blueprintId.?});
+		};
+		const blueprints = blueprintCache.get(blueprintId) orelse {
+			std.log.err("['{s}'] Could not find blueprint '{s}'.", .{stringId, blueprintId});
 			return error.MissingBlueprint;
-		}
-		var self = StructureBuildingBlock{
+		};
+		const self = StructureBuildingBlock{
 			.children = arenaAllocator.alloc(AliasTable(Child), childBlockStringId.items.len),
-			.blueprint = blueprintRef.?,
+			.blueprints = blueprints,
 		};
 		const childrenZon = zon.getChild("children");
 		for(childBlockStringId.items, 0..) |colorName, colorIndex| {
-			self.children[colorIndex] = initChildTableFromZon(stringId, colorName, colorIndex, childrenZon.getChild(colorName));
+			self.children[colorIndex] = try initChildTableFromZon(stringId, colorName, colorIndex, childrenZon.getChild(colorName));
 		}
 		return self;
 	}
 	pub fn getBlueprint(self: StructureBuildingBlock, rotation: Degrees) *BlueprintEntry {
-		return self.blueprint[@intFromEnum(rotation)];
+		return self.blueprints[@intFromEnum(rotation)];
```

Comment:
Should be `&self.blueprints...` otherwise this won't compile.

Reply:
Maybe, I can't compile it anyway cause there is no createAssetStringID implementation.